File: IAIso-v5.0/core/iaiso-php/src/Coordination/RedisClient.php
<?php

declare(strict_types=1);

namespace IAIso\Coordination;

/**
 * Structural Redis client interface. Implement this around your favorite
 * Redis library (phpredis, predis/predis, etc.) to plug it into
 * {@see RedisCoordinator}.
 */
interface RedisClient
{
    /**
     * Run a Lua script.
     *
     * @param string[] $keys
     * @param string[] $args
     * @return mixed parsed Redis reply (list, map, string, or int)
     */
    public function eval(string $script, array $keys, array $args): mixed;

    /**
     * HSET key field value [field value ...]
     *
     * @param array<int,array{0:string,1:string}> $fieldValuePairs
     */
    public function hset(string $key, array $fieldValuePairs): void;

    /**
     * HKEYS key
     *
     * @return string[]
     */
    public function hkeys(string $key): array;
}

File: IAIso-v5.0/core/iaiso-java/iaiso-coordination/src/main/java/io/iaiso/coordination/RedisClient.java
package io.iaiso.coordination;

/**
 * Structural Redis client interface. Implement this around your favorite
 * Redis library (Jedis, Lettuce, Spring Data Redis, etc.) to plug it
 * into {@link RedisCoordinator}.
 */
public interface RedisClient {

    /** Run a Lua script. Returns a parsed Redis reply. */
    Object eval(String script, String[] keys, String[] args);

    /** HSET key field value [field value ...] */
    void hset(String key, String[][] fieldValuePairs);

    /** HKEYS key */
    java.util.List<String> hkeys(String key);
}
